workspaces/ballerina/type-diagram/src/components/entity-relationship/EntityNode/EntityHead/EntityHead.tsx

```diff
@@ -116,6 +131,49 @@ export function EntityHeadWidget(props: ServiceHeadProps) {
         setAnchorEl(null);
     }
 
+    // Keep this sync to satisfy Menu Item's onClick: () => void
+    const handleDeleteClick = () => {
+        const idLabel = node.getID();
+
+        // Position the confirm near the menu button
+        if (anchorEl) {
+            const rect = anchorEl.getBoundingClientRect();
+            const spaceBelow = window.innerHeight - rect.bottom;
+            const spaceAbove = rect.top;
+
+            if (spaceBelow < 200 && spaceAbove > 200) {
+                setAnchorOrigin({ vertical: "top", horizontal: "left" });
+                setTransformOrigin({ vertical: "bottom", horizontal: "right" });
+            } else {
+                setAnchorOrigin({ vertical: "bottom", horizontal: "left" });
+                setTransformOrigin({ vertical: "top", horizontal: "right" });
+            }
+            setConfirmEl(anchorEl);
+        }
+        setAnchorEl(null); // close menu
+
+        // Open confirm immediately in "verifying" mode
+        setIsVerifyingDelete(true);
+        setConfirmMessage("Checking usages...");
+        setIsConfirmOpen(true);
+
+        // Verify asynchronously and update the message
+        verifyTypeDelete(idLabel)
+            .then((canDelete) => {
+                setConfirmMessage(
+                    canDelete
+                        ? `Are you sure you want to delete ${idLabel}?`
+                        : `${idLabel} has usages. Are you sure you want to delete this?`
```

Comment:
How about we update the msg to say it will cause errors
```suggestion
                        : `${idLabel} has usages. Deleting it may cause errors. Are you sure you want to proceed?`
```